### DISCUSSION TITLE
New/Repeat Customer in Amazon SP API

### DISCUSSION THREAD

**Discussion by NitinBhasneria01** [amzn/selling-partner-api-models#3696] [General] [answered]
We want to calculate Customer Retention Rate as, it is one of the most popular metrics that users track on Merito. For which we need to know if an order is from a new customer or a repeat customer. We need to know this information for each order

The logic we're currently using is - looking at the total number of orders from a unique anonymised email ID to ascertain if the user is first time or a repeat user

Here's that approach in 3 steps -
1. GET Orders Tracking Report: 
                GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL
                Rate: 1 API Call for 1 month

2. GET /orders/v0/orders/{orderId}/buyerInfo: 
                We get all the orderIds from step 1 and for each order we fetch the buyer email. We have to use this extra API beacuse in the order's tracking report, buyerEmail is not present.
                Rate: 1 API Call for 1 Order

3. GET /orders/v0/orders: 
                All all the buyerEmails (which we got from step 2), we fetch all the orders and take the orders count. To know how many of the orders are made from that particular customer. 
                Rate: 1 API Call for 1 Email

Comparing the order counts from step 3 to the order counts from Step 1, we get to know whether the customer is new or repeat.

Here are a few questions regarding this - 
1. Is the anonymised email ID we receive in step 2 unique to an individual customer? For eg - is there a scenario which the same customer may have 2 different anonymised email IDs in step 2?
2. Can you please tell us whether this is the ideal/scalable way for ascertaining if a user is first-time or repeat? Please suggest an alternative that can help us improve/decrease the number of API Calls. Please note we have more than 70 brands using Merito.
3. Can we make a Bulk API call on step 2 and 3 to reduce the number of API calls?

**Comment by leonardonakagawa** [comment, marked as answer]
Hi @NitinBhasneria01 

Thank you for reaching out regarding Customer Retention Rate calculation.

1. Yes, email is unique per buyer-seller relationship. Sellers can use the anonymized email alias the same way as a real email address. Furthermore, the anonymized email alias remains the same for all future contact with the same buyer. Buyers receive assigned a unique email alias for each seller relationship. So at a seller level you can track repeat buyers.

For the workflow you are mentioning, I have some suggestions to it:

2. You can leverage on the `GET_FLAT_FILE_ORDER_REPORT_DATA_SHIPPING` report which return the same order information from the one you are using plus the `buyer-email`, with that you don't need to call GET /orders/v0/orders/{orderId}/buyerInfo
 to get buyer email.

You can find details regarding this in [Order reports](https://developer-docs.amazon.com/sp-api/docs/report-type-values-order#order-reports) and [GET_FLAT_FILE_ORDER_REPORT_DATA_SHIPPING](https://developer-docs.amazon.com/sp-api/docs/order-reports-attributes#get_flat_file_order_report_data_shipping) to see the report attributes.

If this report is suitable for your use case, this change will help decreasing the number of your API calls. You can also try to only use reports as the one I mentioned returns the email address.

Thanks,
Leonardo Nakagawa
Solutions Architect, Selling Partner API Developer Support
